io/GDALWriter.cpp

```diff
@@ -321,6 +329,45 @@ void GDALWriter::doneFile()
         throwError(raster.errorMsg());
 
     getMetadata().addList("filename", m_filename);
+
+    std::vector<std::string> gdalitems = Utils::split(m_GDAL_metadata, ',');
+    for (auto& v: gdalitems)
+    {
+        const std::size_t pos = v.find_first_of("=");
+        if (pos != std::string::npos)
+        {
+            const std::string name = v.substr(0, pos);
+            const std::string value = pos != std::string::npos ? v.substr(pos + 1) : "";
```

Comment:
Is it really OK to have empty metadata values?

Reply:
conflicted on this. It is tested in the tests, btw.

It is possible that GDAL gives us an empty item, and I wanted to be able to support it, or users might wonder where their keys went.